jenkins/scripts/node-test-commit-pre.sh

```diff
@@ -27,6 +27,13 @@ git status
 git rev-parse HEAD
 git rev-parse $REBASE_ONTO
 
+# COMMIT_SHA_CHECK needs to be set in the job. Check that the gitref
+# that is checked out hasn't been updated since the job was requested.
+if [ "$(git rev-parse HEAD)" != "$(git rev-parse ${COMMIT_SHA_CHECK})" ]; then
+    echo "HEAD does not match expected COMMIT_SHA_CHECK"
+    exit 1
+fi
```

Comment:
We might be able to change the checkout on L20 above to checkout `$COMMIT_SHA_CHECK` but I don't know if anything will break if the checkout is not a checkout of a branch.

Reply:
We can deploy the change and monitor for errors.